code/game/objects/items/weapons/storage/pouch.dm

```diff
@@ -0,0 +1,454 @@
+		return
+
+	else if(length(contents) <= storage_slots * 0.5)
+		overlays += "+[icon_state]_half"
+	else
+		overlays += "+[icon_state]_full"
+
+
+/obj/item/storage/pouch/get_examine_text(mob/user)
+	. = ..()
+	. += "Can be worn by attaching it to a pocket."
+
+/obj/item/storage/pouch/general
+	name = "light general pouch"
+	desc = "A general-purpose pouch used to carry a small item, or two tiny ones."
+	icon_state = "small_drop"
+
+	max_w_class = ITEMSIZE_NORMAL
+	cant_hold = list( //Prevent inventory bloat
+		/obj/item/storage/firstaid,
+		/obj/item/storage/bible,
+		/obj/item/storage/box,
+	)
+	storage_slots = null
+	max_storage_space = 2
+
+/obj/item/storage/pouch/general/medium
+	name = "medium general pouch"
+	desc = "A general-purpose pouch used to carry a variety of differently sized items."
+	icon_state = "medium_drop"
+	storage_slots = null
+	max_storage_space = 4
+
+/obj/item/storage/pouch/general/large
+	name = "large general pouch"
+	desc = "A general-purpose pouch used to carry more differently sized items."
+	icon_state = "large_drop"
+	storage_slots = null
+	max_storage_space = 6
+
+/obj/item/storage/pouch/firstaid
+	name = "first-aid pouch"
+	desc = "It contains, by default, autoinjectors. But it may also hold ointments, bandages, and pill bottles."
+	icon_state = "firstaid"
+	storage_slots = 4
+	can_hold = list(
+		/obj/item/stack/medical/ointment,
+		/obj/item/reagent_containers/hypospray/autoinjector,
+		/obj/item/storage/pill_bottle,
+		/obj/item/stack/medical/bruise_pack,
+		/obj/item/stack/medical/splint,
+	)
+
+/obj/item/storage/pouch/firstaid/full
+	desc = "Contains a painkiller autoinjector, first-aid autoinjector, some ointment, and some bandages."
+
+/obj/item/storage/pouch/firstaid/full/fill()
+	new /obj/item/reagent_containers/hypospray/autoinjector/bicaridine(src)
+	new /obj/item/reagent_containers/hypospray/autoinjector/kelotane(src)
+	new /obj/item/reagent_containers/hypospray/autoinjector/perconol(src)
+	new /obj/item/reagent_containers/hypospray/autoinjector/emergency(src)
+
+/obj/item/storage/pouch/firstaid/full/alternate/fill()
+	new /obj/item/reagent_containers/hypospray/autoinjector/tricord(src)
+	new /obj/item/stack/medical/splint(src)
+	new /obj/item/stack/medical/ointment(src)
+	new /obj/item/stack/medical/bruise_pack(src)
+
+/obj/item/storage/pouch/pistol
+	name = "sidearm pouch"
+	desc = "You could carry a pistol in this; more importantly, you could draw it quickly. Useful for emergencies."
+	icon_state = "pistol"
+	use_sound = null
+	max_w_class = ITEMSIZE_NORMAL
+	can_hold = list(/obj/item/gun)
+
+	flap = FALSE
+
+	/// The gun it holds, used for referencing later so we can update the icon.
+	var/obj/item/gun/current_gun
+
+	/// The underlay we will use for the gun
+	var/image/gun_underlay
+
+	/// The X offset of the gun overlay
+	var/icon_x = 0
+
+	/// The Y offset of the gun overlay
+	var/icon_y = -3
+
+/obj/item/storage/pouch/pistol/Destroy()
+	gun_underlay = null
+	current_gun = null
```

Comment:
you need to either QDEL_NULL it or move it out of the pouch if the pouch is deleted, with a strong preference to the former